mode/sparql/sparql.js

```diff
@@ -72,11 +81,19 @@ CodeMirror.defineMode("sparql", function(config) {
       return "meta";
     }
     else {
-      stream.eatWhile(/[_\w\d]/);
-      if (stream.eat(":")) {
-        eatPnLocal(stream);
-        return "atom";
+      if (PREFIX_START.test(ch)) {
+        if (stream.eatWhile(PREFIX_MID)) {
+          stream.backUp(1);
+          if (stream.eat(PREFIX_END) && stream.eat(":")) {
+            eatPnLocal(stream);
+            return "atom";
+          }
```

Comment:
This falls through if the `eat` returns false, despite having consumed characters. What is it testing for, precisely? And does the set of allowed prefix characters really differ from the set of allowed variable characters?

Reply:
It was intentional to eat then fall through, since the remainder of the method used stream.current() and would include those characters: however there was a bug there in that the characters allowed for variables do not include some of those allowed for prefixes so stream.current() will contain characters that should not be styled cm-variable. 

I have rewritten the change to use a more complex regular expression for matching every character after 'ch' (with the existing code for the edge case where there is nothing before the colon left untouched).